src/main.rs

```diff
@@ -1,167 +1,8 @@
-#[macro_use]
-extern crate diesel_migrations;
-
-use actix_web::{middleware, web::Data, App, HttpServer, Result};
-use diesel_migrations::EmbeddedMigrations;
-use doku::json::{AutoComments, CommentsStyle, Formatting, ObjectsStyle};
-use lemmy_api_common::{
-  context::LemmyContext,
-  lemmy_db_views::structs::SiteView,
-  request::build_user_agent,
-  utils::{
-    check_private_instance_and_federation_enabled,
-    local_site_rate_limit_to_rate_limit_config,
-  },
-  websocket::chat_server::ChatServer,
-};
-use lemmy_db_schema::{
-  source::secret::Secret,
-  utils::{build_db_pool, get_database_url, run_migrations},
-};
-use lemmy_routes::{feeds, images, nodeinfo, webfinger};
-use lemmy_server::{
-  api_routes_http,
-  code_migrations::run_advanced_migrations,
-  init_logging,
-  root_span_builder::QuieterRootSpanBuilder,
-  scheduled_tasks,
-};
-use lemmy_utils::{
-  error::LemmyError,
-  rate_limit::RateLimitCell,
-  settings::{structs::Settings, SETTINGS},
-};
-use reqwest::Client;
-use reqwest_middleware::ClientBuilder;
-use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
-use reqwest_tracing::TracingMiddleware;
-use std::{env, sync::Arc, thread, time::Duration};
-use tracing_actix_web::TracingLogger;
-
-pub const MIGRATIONS: EmbeddedMigrations = embed_migrations!();
```

Comment:
Weird how rust didn't check the stuff in main, but I noticed that on my system too.

Reply:
Its because the item is marked as public, so Rust will never mark it as unused as it might be used by another crate. Unfortunately there is no way to change this logic, so there are probably more public unused items across Lemmy codebase. The only tool I found to detect unused code in workspaces gave only false positives.